src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -41,6 +41,42 @@ public static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        private static ReadOnlySpan<uint> CrcTable => new uint[]
```

Comment:
Is this good practise or is using a static heap allocated array a better alternative?

Reply:
This will allocate the `uint[]` array on each access. The Roslyn optimization (if you had that in mind) only works for one byte sized primitive types at the moment (due to endianess concerns).

> static heap allocated array a better alternative?

At the moment yes, so
```suggestion
        private static readonly uint[] s_crcTable => new uint[]
```

Reply:
Is it still the rule to use the prefix `_s` in non-legacy code? And would stack allocation still make sense?
Something like: ```cs
private static ReadOnlySpan<uint> CrcTable => s_crcTable;
```

Reply:
To which stack allocation do you refer? Here's a GC-heap allocation, and that's fine (here) as it's a one-timer (due static readonly).

Allocating the constants on the stack requires a lot of stack spaces, and would be needed to be done for each invocation of the method (or at least for the software fallback) and is not very efficient.

Reply:
👍🏼